### PR TITLE
Rename `transform_to_shard_dict` api to `shardsPreprocessing` and other small improvements

### PR DESCRIPTION
**Pull request by dding3**
## Description

Rename `transform_to_shard_dict` api to `shardsPreprocessing` and add documents about this api.
Remove rdd operations in pytorch tutorials

### 1. Why the change?

Fix comments in https://github.com/intel-analytics/BigDL/pull/5077#pullrequestreview-1055487451

### 2. User API changes

Rename `transform_to_shard_dict` api to `shardsPreprocessing`, no parameters change

### 3. Summary of the change 

Rename `transform_to_shard_dict` api to `shardsPreprocessing` and add documents about this api.
Remove rdd operations in pytorch tutorials

### 4. How to test?
-  Application test

### CODE REVIEW

File: python/orca/example/shard_keras_tutorial.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 # compile the keras model
 model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
 
-data_shard = transform_to_shard_dict(data_shard,
+data_shard = shardsPreprocessing(data_shard,
```
Comment (jason-dai) [inline python/orca/example/shard_keras_tutorial.py:40]: "Preprocessing" is a very general term; be more specific about how the shards are transformed

File: python/orca/src/bigdl/orca/data/utils.py [hunk 2]
```diff
@@ -467,7 +467,15 @@ def check_col_exists(df, columns):
                           str(col_not_exist) + " do not exist in this Table")
 
 
-def transform_to_shard_dict(data, featureCols, labelCol):
+def shardsPreprocessing(data, featureCols, labelCol):
+    """
+    This api is used to process SparkXShards of pandas dataframe to SparkXShards of
+    dictionary, x is feature, y is label
```
Comment (jason-dai) [inline python/orca/src/bigdl/orca/data/utils.py:473]: it is not clear what exactly it does; show some examples in the description

File: python/orca/src/bigdl/orca/data/utils.py [hunk 3]
```diff
@@ -467,7 +467,15 @@ def check_col_exists(df, columns):
...
+    :param data: SparkXShards of pandas dataframe.
+    :param featureCols: a list of featurecols.
+    :param labelCol: label col.
```
Comment (jason-dai) [inline python/orca/src/bigdl/orca/data/utils.py:476]: Be specific about the types of `featureCols` and `labelCol`. Can `labelCol` be a list?

**Comment by dding3** [comment]
most jekins pass. http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-Orca-PR-Validation/882/. Failed one is not related to this pr.

File: python/orca/src/bigdl/orca/data/utils.py [hunk 4]
```diff
@@ -467,7 +467,22 @@ def check_col_exists(df, columns):
                           str(col_not_exist) + " do not exist in this Table")
 
 
-def transform_to_shard_dict(data, featureCols, labelCol):
+def shards_pd_df_to_shards_dic(data, featureCols, labelCol):
```
Comment (jason-dai) [inline python/orca/src/bigdl/orca/data/utils.py:470]: The name is still very generic; what does this mean exactly? A user will not be able to understand what `shards_pd_df_to_shards_dic` is at all.

What are the valid types of each column? Float or array or ...?

Do you assembly these columns into a vector or a tensor or ...? If so, make it explicit in the name.

File: python/orca/src/bigdl/orca/data/utils.py [hunk 5]
```diff
@@ -467,7 +467,22 @@ def check_col_exists(df, columns):
...
+    This api is used to process SparkXShards of pandas dataframe to SparkXShards of
+    dictionary, x is feature, y is label
+    :param data: SparkXShards of pandas dataframe.
+    :param featureCols: a list of featurecols.
+    :param labelCol: single label col.
+    :return: SparkXShards of dictionary
+
+    eg:
+    shards: SparkXShards of pandas dataframe with 2 cols ['f1', 'lable'],
+    transform_shards = shards_pd_df_to_shards_dic(shards, featureCols=[''], labelCol='')
```
Comment (jason-dai) [inline python/orca/src/bigdl/orca/data/utils.py:481]: Show this example graphically

**Pull request merged**